notebooks/Offline Checks.ipynb

```diff
@@ -0,0 +1,813 @@
+    "\n",
+    "        elif op_layer[\"layerType\"] == \"ArcGISFeatureLayer\":\n",
+    "            layer_url = op_layer[\"url\"] if \"url\" in op_layer else op_layer.url\n",
+    "            layer_title = op_layer[\"title\"] if \"title\" in op_layer else op_layer.title\n",
+    "            feature_layer = FeatureLayer(layer_url)\n",
+    "            if not is_sync_enabled(feature_layer):\n",
+    "                layer_errors.append(f\"{layer_title} is sync not enabled\")\n",
+    "            if not is_supported_feature_layer_field(feature_layer.properties.fields):\n",
+    "                layer_errors.append(f\"{layer_title} has field name errors\")\n",
+    "            if not is_supported_feature_layer_indices(feature_layer):\n",
+    "                layer_errors.append(f\"{layer_title} has unsupported layer indicies\")\n",
+    "            if is_join_view(feature_layer):\n",
+    "                layer_errors.append(f\"{layer_title} is a join view\")\n",
+    "            if is_global_id_missing(feature_layer):\n",
+    "                layer_errors.append(f\"{layer_title} has globalid missing\")\n",
+    "            if is_relationship_missing(feature_layer):\n",
+    "                layer_errors.append(\n",
+    "                    f\"{layer_title} has missing relationship field names\"\n",
+    "                )\n",
+    "            if is_subtype_missing(feature_layer):\n",
+    "                layer_errors.append(f\"{layer_title} has missing subtype field\")\n",
+    "            if not is_true_curve_ready(feature_layer):\n",
+    "                layer_errors.append(f\"{layer_title} is not true curve enabled\")\n",
+    "            # add more tests here\n",
+    "        elif op_layer[\"layerType\"] == \"GroupLayer\":\n",
+    "            layer_errors.extend(check_layers(op_layer[\"layers\"]))\n",
+    "        elif op_layer[\"layerType\"] in [\n",
+    "            \"ArcGISImageServiceLayer\",\n",
+    "            \"ArcGISTiledImageServiceLayer\",\n",
+    "            \"ArcGISTiledMapServiceLayer\",\n",
+    "            \"VectorTileLayer\",\n",
+    "        ]:\n",
+    "            layer_title = op_layer[\"title\"] if \"title\" in op_layer else op_layer.title\n",
+    "            if not is_export_enabled(op_layer):\n",
+    "                layer_errors.append(f\"{layer_title} export tiles not enabled\")\n",
+    "            else:\n",
+    "                if is_map_multisource(op_layer):\n",
+    "                    layer_errors.append(f\"{layer_title} is multisource\")\n",
+    "                if is_deprecated_basemap_urls(op_layer):\n",
+    "                    layer_errors.append(f\"{layer_title} is depcrecated\")\n",
+    "\n",
+    "    layer_errors = list(dict.fromkeys(layer_errors))\n",
+    "\n",
+    "    return layer_errors"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 157,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def check_tables(\n",
```

Comment:
Extra whitespace/return and comma here? should be `check_tables( tables) -> str:`